### ISSUE TITLE
No idea what the API Token is

### ISSUE THREAD

**Issue by dheath73** [andrewsayre/hass-smartthings-remove#1]
Hello tiny little issue here.
Due to the problems I had getting Smartthings to rework with Hassio 0.89 I thought I would try a new Smartthings API token. However I didn't save the old one. Now I cannot remove the SmartApp from the Smartthings Classic App and I still cannot get the intergration to work between SmartThings and Hassio.

Any suggestion please?

Thank you.

**Comment by andrewsayre** [comment]
As long as you're using the same ST/Samsung account, it doesn't matter what token you used previously.  Just create a new one (and remove the old one from the portal) to run this utility.

If you're looking for help with the HASS integration itself, please open an issue there: https://github.com/home-assistant/home-assistant/issues

**Comment by dheath73** [comment]
Thanks for your advice. Whats weird is one  of the home assistants smart apps will not remove. I've tried three times to try and work out what is happening and although it remove the smart app associated to the API Token as expected, however one is always left behind. It's been over a day so I don't think its the sync issue.

**Comment by andrewsayre** [comment]
Does the utility output anything when you run it?  If you don't see anything, it's not finding anything matching in the account.  Some possibilities:
1) You actually used a different ST/Samsung account with the original token.
1) There's a data/sync issue with the account or ST location

**Comment by dheath73** [comment]
this is the output when I run it 

Traceback (most recent call last):
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Anderson\AppData\Local\Programs\Python\Python37-32\Scripts\hass_smartthings_remove.exe\__main__.py", line 9, in <module>
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\site-packages\hass_smartthings_remove\remove.py", line 16, in main
    loop.run_until_complete(remove_apps(args.token))
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\asyncio\base_events.py", line 584, in run_until_complete
    return future.result()
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\site-packages\hass_smartthings_remove\remove.py", line 25, in remove_apps
    apps = await api.apps()
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\site-packages\pysmartthings\smartthings.py", line 89, in apps
    resp = await self._service.get_apps(params)
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\site-packages\pysmartthings\api.py", line 156, in get_apps
    return await self.get_items(API_APPS, params=params)
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\site-packages\pysmartthings\api.py", line 375, in get_items
    'get', self._api_base + resource, params, None)
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\site-packages\pysmartthings\api.py", line 366, in request
    resp.raise_for_status()
  File "c:\users\anderson\appdata\local\programs\python\python37-32\lib\site-packages\aiohttp\client_reqrep.py", line 942, in raise_for_status
    headers=self.headers)
aiohttp.client_exceptions.ClientResponseError: 401, message='Unauthorized'

**Comment by andrewsayre** [comment]
The token you're using is invalid.  Only reason you get a 401 error.

**Comment by dheath73** [comment]
Sorry for the delay in replying. 
This morning I have tried this again and have done as you suggested. I created a new api token. I then run the command with the new API Token. However my previous instance of the SmartThings App still remains. I only have the one smartThings account so what you are suggesting should remove the original SmartThings App which was installed prior to the update to Hassio 0.89 ?? 

Are there any other suggestions you could think of for me to try please?

Thank you for your help with this, I do really appreciate it.

**Comment by dheath73** [comment]
Morning. I still have the old smartthings app and its still interfering with me getting the information from smartthings to Home assistant. Can you suggest any method of how I can remove this old smart app please? 
As you can see I only have one API listed
![download (2)](https://user-images.githubusercontent.com/34939165/54347413-e414dd00-463e-11e9-96b0-e3e3ec5869b7.png)

And this is my Smartthings app on my phone showing the two apps. 
![e8a8d33d-616f-42d6-ac8c-3796c71ee251](https://user-images.githubusercontent.com/34939165/54347564-25a58800-463f-11e9-8c11-a636d805f2b8.jpg)

Just to clarify no matter how many times I run "hass_smartthings_remove " with my new API that I have generated it will not remove the old Home assistant smart app (called Smart Home Monitor) from my smartthings classic app.

Do you have any suggestions please?

Thank you

**Comment by dheath73** [comment]
I try to delete this rogue app via the new smarthings App and it deleted it! Thanks for everyone who read this and for all your help.

**Comment by andrewsayre** [comment]
> try this via the new smarthings App and it deleted it! Thanks for everyone who read this and for all your help.

Interesting... 👍 glad that worked.

**Comment by dheath73** [comment]
Just so you know I am also able to delete the new app in there as well so there maybe no need to use your script for anyone else reading this.

To delete the rogue app I installed and opened the new SmartThings App. Then I tapped the 3 horizontal lines > Automatons > 3 dots > Delete > held my finger on the Home Assistant App I wanted to delete to select it > Delete and Delete again to confirm.

Hope this may help someone else.
